File: gang/week6/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        long n = num;       // num의 범위가 int형보다 크기 때문에 long형으로 바꿈
        while(n != 1) {     // n이 1이 아닐때까지 반복
            if(n % 2 == 0) {
                n /= 2;
            }
            else {
                n = n * 3 + 1;
            }
            answer++;

            if(answer == 500) {     // 횟수가 500번이면 -1 리턴
                return -1;
            }
        }
        return answer;
    }
}

File: jeonghyeon/6week/콜라츠 추측.js
function solution(num) {
    let answer = 0; // 몇번 반복하는지 셈
    while(num != 1) {
        if(num % 2 == 0) { // 입력된 수가 짝수
            num /= 2; // 2로 나눠진 수
        } else { // 홀수 시
            num = num * 3 + 1;
        }
        answer += 1 // 반복한 수 
    }
    if(answer >= 500) {
        return -1; // 500번 반복 후 1 안나오면 -1
    } else {
        return answer; // 아니면 반복한 수 리턴
    }
}